### PR TITLE
Update minimum Python version to Python 3.9

### PR DESCRIPTION
**Pull request by shwina**

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -62,13 +62,13 @@ jobs:
       run:
         shell: bash
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@py-39
```
Comment (ajschmidt8) [inline .github/workflows/build.yaml:65]:
```suggestion
      - uses: actions/checkout@v3
```

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -62,13 +62,13 @@ jobs:
...
         with:
           fetch-depth: 0
       - name: Build wheel
         run: ci/build_python_pypi.sh
       - name: Publish distribution 📦 to PyPI
         if: inputs.build_type == 'nightly'
-        uses: pypa/gh-action-pypi-publish@release/v1
+        uses: pypa/gh-action-pypi-publish@py-39
```
Comment (ajschmidt8) [inline .github/workflows/build.yaml:72]:
```suggestion
        uses: pypa/gh-action-pypi-publish@release/v1
```

File: .github/workflows/pr.yaml [hunk 3]
```diff
@@ -53,7 +53,7 @@ jobs:
       run:
         shell: bash
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@py-39
```
Comment (ajschmidt8) [inline .github/workflows/pr.yaml:56]:
```suggestion
      - uses: actions/checkout@v3
```

**Review by ajschmidt8 — COMMENTED** [review]
whoops. looks like my `sed` command was too aggressive.

for the future, this revised one will ensure that only lines containing the string `shared-action-workflows` are targeted.

```sh
sed -i '/shared-action-workflows/ s/@.*/@py-39/g;' ./.github/workflows/*.yaml
```

I'll manually update this PR

**Review by ajschmidt8 — APPROVED** [review]

**Comment by ajschmidt8** [comment]
Seems like this conditional statement can be removed now:

https://github.com/rapidsai/dask-cuda/blob/9dbfd1c936f8a60b42bb275bd73d201d71b606b6/dask_cuda/device_host_file.py#L243-L269

I'll open an issue for this for a follow-up PR.

**Comment by ajschmidt8** [comment]
queuing to merge upon Python codeowner approval

File: pyproject.toml [hunk 4]
```diff
@@ -33,7 +33,7 @@ classifiers = [
     "Topic :: Scientific/Engineering",
     "License :: OSI Approved :: Apache Software License",
     "Programming Language :: Python :: 3",
-    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
```
Comment (charlesbluca) [inline pyproject.toml:36]:
```suggestion
```

**Review by charlesbluca — APPROVED** [review]
Thanks @shwina and @ajschmidt8!

**Review by pentschev — APPROVED** [review]
Thanks @shwina for the work in making sure we don't fall behind the rest of the PyData community on Python version support! 😄

**Pull request merged**